packages/taiko-client/driver/chain_syncer/blob/blocks_inserter/pacaya.go

```diff
@@ -0,0 +1,379 @@
+					if err != nil {
+						return fmt.Errorf("failed to fetch last batch (%d): %w", meta.GetBatchID().Uint64()-1, err)
+					}
+					parentNumber = new(big.Int).SetUint64(lastBatch.LastBlockId)
+				}
+			} else {
+				parentNumber = new(big.Int).SetUint64(lastPayloadData.Number)
+			}
+
+			parent, err = i.rpc.L2ParentByCurrentBlockID(ctx, new(big.Int).Add(parentNumber, common.Big1))
+		}
+		if err != nil {
+			return fmt.Errorf("failed to fetch L2 parent block: %w", err)
+		}
+
+		log.Debug(
+			"Parent block",
+			"blockID", parent.Number,
+			"hash", parent.Hash(),
+			"beaconSyncTriggered", i.progressTracker.Triggered(),
+		)
+
+		blockID := new(big.Int).SetUint64(parent.Number.Uint64() + 1)
+		difficulty, err := encoding.CalculatePacayaDifficulty(blockID)
+		if err != nil {
+			return fmt.Errorf("failed to calculate difficulty: %w", err)
+		}
+		timestamp := meta.GetLastBlockTimestamp()
+		for i := len(meta.GetBlocks()) - 1; i >= 0; i-- {
+			timestamp = timestamp - uint64(meta.GetBlocks()[i].TimeShift)
+		}
+
+		baseFee, err := i.rpc.CalculateBaseFee(
+			ctx,
+			parent,
+			true,
+			meta.GetBaseFeeConfig(),
+			timestamp,
+		)
+		if err != nil {
+			return err
+		}
+
+		log.Info(
+			"L2 baseFee",
+			"blockID", blockID,
+			"baseFee", utils.WeiToGWei(baseFee),
+			"parentGasUsed", parent.GasUsed,
+			"batchID", meta.GetBatchID(),
+			"indexInBatch", j,
+		)
+
+		// Assemble a TaikoAnchor.anchorV3 transaction
+		anchorBlockHeader, err := i.rpc.L1.HeaderByHash(ctx, meta.GetAnchorBlockHash())
+		if err != nil {
+			return fmt.Errorf("failed to fetch anchor block: %w", err)
+		}
+		anchorTx, err := i.anchorConstructor.AssembleAnchorV3Tx(
+			ctx,
+			new(big.Int).SetUint64(meta.GetAnchorBlockID()),
+			anchorBlockHeader.Root,
+			meta.GetAnchorInput(),
+			parent.GasUsed,
+			meta.GetBaseFeeConfig(),
+			meta.GetSignalSlots(),
+			new(big.Int).Add(parent.Number, common.Big1),
+			baseFee,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create TaikoAnchor.anchorV3 transaction: %w", err)
+		}
+
+		// Decompress the transactions list and try to insert a new head block to L2 EE.
+		if lastPayloadData, err = createPayloadAndSetHead(
+			ctx,
+			i.rpc,
+			&createPayloadAndSetHeadMetaData{
+				createExecutionPayloadsMetaData: &createExecutionPayloadsMetaData{
+					BlockID:               blockID,
+					ExtraData:             meta.GetExtraData(),
+					SuggestedFeeRecipient: meta.GetCoinbase(),
+					GasLimit:              uint64(meta.GetGasLimit()),
+					Difficulty:            common.BytesToHash(difficulty),
+					Timestamp:             timestamp,
+					ParentHash:            parent.Hash(),
+					L1Origin: &rawdb.L1Origin{
+						BlockID:       blockID,
+						L2BlockHash:   common.Hash{}, // Will be set by taiko-geth.
+						L1BlockHeight: meta.GetRawBlockHeight(),
+						L1BlockHash:   meta.GetRawBlockHash(),
+					},
+					Txs:         allTxs[txListCursor : txListCursor+int(blockInfo.NumTransactions)],
```

Comment:
testing on our Helder devnet, if you submit a garbage block with a calldata txList and a block with numTransactions 1 (but the calldata txList is just garbage), this panics.
```go
panic: runtime error: slice bounds out of range [:1] with capacity 0

goroutine 548 [running]:
github.com/taikoxyz/taiko-mono/packages/taiko-client/driver/chain_syncer/blob/blocks_inserter.(*BlocksInserterPacaya).InsertBlocks(0xc000a02050, {0x284d118, 0xc00052b0e0}, {0x285e560, 0xc0000fc6c8}, 0xc0018f4500, 0xc00052b0e0?)
	/build/packages/taiko-client/driver/chain_syncer/blob/blocks_inserter/pacaya.go:210 +0x1dfe
```

Reply:
fixed: https://github.com/taikoxyz/taiko-mono/pull/18746/commits/8552ab4d6a10cfb4e26dec022a4bfdd6e5295a33